contracts/governance/extensions/GovernorStorage.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.19;
+
+import "../Governor.sol";
+
+/**
+ * @dev Extension of {Governor} that implements storage of proposal details. This modules also provides primitives for
+ * the enumerability of proposals.
+ *
+ * Usecases for this module include:
+ * - UIs that expore the proposal state without relying on event indexing.
+ * - For some L2 chains where storage is compared to calldata, using a the proposalId version of queue and execute
+ *   might me cheaper.
+ *
+ * _Available since v5.0._
+ */
+abstract contract GovernorStorage is Governor {
+    struct ProposalDetails {
+        address[] targets;
+        uint256[] values;
+        bytes[] calldatas;
+        bytes32 descriptionHash;
+    }
+
+    uint256[] private _proposalIds;
+    mapping(uint256 => ProposalDetails) private _proposalDetails;
+
+    /**
+     * @dev See {IGovernor-propose}.
+     */
+    function propose(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        string memory description
+    ) public virtual override returns (uint256) {
+        uint256 proposalId = super.propose(targets, values, calldatas, description);
+        // store
+        _proposalIds.push(proposalId);
+        _proposalDetails[proposalId] = ProposalDetails({
+            targets: targets,
+            values: values,
+            calldatas: calldatas,
+            descriptionHash: keccak256(bytes(description))
+        });
+        return proposalId;
+    }
+
+    /**
+     * @dev ProposalId version of {IGovernorTimelock-queue}.
+     */
+    function queue(uint256 proposalId) public virtual {
+        ProposalDetails storage proposalDetails = _proposalDetails[proposalId];
+        queue(
+            proposalDetails.targets,
+            proposalDetails.values,
+            proposalDetails.calldatas,
+            proposalDetails.descriptionHash
+        );
+    }
+
+    /**
+     * @dev ProposalId version of {IGovernor-execute}.
+     */
+    function execute(uint256 proposalId) public payable virtual {
+        ProposalDetails storage proposalDetails = _proposalDetails[proposalId];
+        execute(
+            proposalDetails.targets,
+            proposalDetails.values,
+            proposalDetails.calldatas,
+            proposalDetails.descriptionHash
+        );
+    }
+
+    /**
+     * @dev ProposalId version of {IGovernor-cancel}.
+     */
+    function cancel(uint256 proposalId) public virtual {
+        ProposalDetails storage proposalDetails = _proposalDetails[proposalId];
+        cancel(
+            proposalDetails.targets,
+            proposalDetails.values,
+            proposalDetails.calldatas,
+            proposalDetails.descriptionHash
+        );
+    }
+
+    /**
+     * @dev Return the number of stored proposals.
+     */
+    function getProposalCount() public view virtual returns (uint256) {
```

Comment:
What's the rationale for naming the getters as `get`? We're naming other proposal getters as just `proposal*`, in this case, it'd be `proposalCount()`.

Non-blocking either.

Reply:
I'm open to renaming them.

Reply:
It seems like we should remove the `get` prefixes if we want to be consistent with the rest of the interface. I don't feel super strongly about this but I would do it.

Reply:
Also not super strong but I think a common feature of Solidity is having a getter for every public variable. I'd argue that the developers are used to not seeing the `get` prefix.

One could argue we shouldn't do Solidity specifics, but this doesn't affect external contracts and we're indeed coding it in Solidity.

Reply:
Renamed without `get`.